assets/js/components/CallToAction.js

```diff
@@ -1,77 +1,20 @@
 'use strict';
 
-const utils = require('../libs/elife-utils')();
+const Dismissible = require('./_Dismissible');
 
 module.exports = class CallToAction {
   constructor($elm, _window = window, doc = document) {
     this.$elm = $elm;
     this.window = _window;
     this.doc = doc;
 
-    this.cookieName = CallToAction.deriveCookieName(this.$elm);
-    if (!this.cookieName.length || this.hasAlreadyBeenSeen(this.cookieName)) {
-      this.hide();
-      return;
-    }
-
-    this.$button = CallToAction.buildDismissButton();
-    this.$elm.addEventListener('click', this.handleInteraction.bind(this));
+    this.dismissible = new Dismissible(this.$elm, '.call-to-action', this.doc);
+    this.$elm.addEventListener('click', () => this.dismissible.dismiss());
```

Comment:
doesn't this duplicate https://github.com/elifesciences/pattern-library/pull/958/files#diff-b24b40c83ce03b33cefdfd2daffdc65eR56 where the button has its own event listener?
Also, does it mean that clicking anywhere on the CallToAction triggers the dismissal?

Reply:
This should be attached to the button of the call to action pattern, rather than the entire thing, to dismiss it once it's actioned. You're right that currently it is dismissed too keenly, when any part of the pattern is clicked on.

Reply:
Updated in https://github.com/elifesciences/pattern-library/pull/958/commits/da42b336745f9fd75147d9a705f4f13998043c8e